go.mod

```
@@ -1,114 +1,115 @@
 module github.com/kubeflow/mpi-operator
 
-go 1.22.0
+go 1.23.0
+
+toolchain go1.23.1
```

```suggestion
go 1.23
```

AFAIK, since Go 1.23, we do not need to specify the toolchain version.